tests/abi/tools/revng-abi-verify/Verify.cpp

```diff
@@ -7,374 +7,228 @@
-              return;
-            }
-            for (size_t Index = 0; Index < ArgumentCount; ++Index) {
-              if (!areArgumentsCompatible(Iteration.Arguments[Index],
-                                          Upcasted.Arguments.at(Index),
-                                          RegisterSize)) {
-                size_t ItID = std::distance(CurrentArtifact->Iterations.begin(),
-                                            &Iteration);
-                std::string_view NmView = llvm::StringRef(Upcasted.CustomName);
-                dbg << "Argument " << Index << " is not compatible with "
-                    << "the expected type on the iteration #" << ItID
-                    << " of a function named \"" << NmView << "\":\n";
-                Upcasted.dump();
-                Result = ExitCode::FailedArgumentCompatibilityCheck;
-                return;
-              }
-            }
-          }
-        }
-      } else if constexpr (std::is_same_v<UpcastedType, RawFunctionType>) {
-        auto CurrentArtifact = ParsedArtifact.find(Upcasted.CustomName);
-        if (CurrentArtifact != ParsedArtifact.end()) {
-          for (auto &Argument : Upcasted.Arguments) {
-            auto Indices = verifyRegister(Argument.Location,
-                                          *CurrentArtifact,
-                                          RegisterSize);
-            if (Indices.empty()) {
-              std::string_view NameView = llvm::StringRef(Upcasted.CustomName);
-              dbg << "Cannot locate an argument in the artifact:\n";
-              Argument.dump();
-              dbg << "The function named \"" << NameView << "\":\n";
-              Upcasted.dump();
-              Result = ExitCode::FailedLocatingAnArgument;
-              return;
-            }
-
-            if (Indices.size() > 1) {
-              std::string_view NameView = llvm::StringRef(Upcasted.CustomName);
-              dbg << "Multiple locations for an argument in the artifact:\n";
-              Argument.dump();
-              dbg << "The locations are: ";
-              for (auto Index : Indices)
-                dbg << Index << ' ';
-              dbg << "\nThe function named \"" << NameView << "\":\n";
-              Upcasted.dump();
-              Result = ExitCode::FailedSelectingSingleArgumentLocation;
-              return;
-            }
-          }
-        }
-      }
-    };
-    Type.upcast(Visitor);
-    if (Result != ExitCode::Success)
-      return Result;
+    revng_assert(Type.get() != nullptr);
+
+    auto CurrentArtifact = ParsedArtifact.find(Type->name());
+    if (CurrentArtifact == ParsedArtifact.end()) {
+      // Ignore types with no artifact provided.
+      continue;
+    }
+
+    if (auto *CABI = llvm::dyn_cast<model::CABIFunctionType>(Type.get())) {
+      if (auto Error = verifyImpl(abi::FunctionType::Layout(*CABI),
+                                  *CurrentArtifact,
+                                  ABI))
+        return Error;
```

Comment:
Why not simply...
```suggestion
      return verifyImpl(abi::FunctionType::Layout(*CABI),
                                  *CurrentArtifact,
                                  ABI);
```
?
Alsom you could keep in the `if` only the computation of the layout and have a single invocation to `verifyImpl`. No biggie though.

Reply:
It's inside a loop. I want to drop down if there's an error but keep going if not.

Reply:
Aahh right